MediaRecorder.html

```diff
@@ -442,6 +443,38 @@
               <em>Return type:</em> <code>boolean</code>
             </div>
           </dd>
+
+          <dt><dfn><code>getSupportedTypes</code></dfn>, static</dt>
+          <dd>
+            When <code>getSupportedTypes()</code> method is invoked, the UA MUST
+            queue a task, using the DOM manipulation task source, that runs the
+            following steps:
+            <ol>
+            <li>Let <var>p</var> be a promise.</li>
+            <li>Run the following steps in parallel:
+             <ol>
+             <li>Let <var>resultList</var> be an empty list.
+             <li>For each <a href="https://tools.ietf.org/html/rfc2046">MIME
+             type</a> <var>type</var> that is supported for encoding and/or
+             multiplexing (i.e. <code>isSuportedType(<var>mimeType</var>)</code>
+             is true), add <var>type</var> to <var>resultList</var>.
+             </li>
+             <li>Resolve <var>p</var> with <var>resultList</var>.</li>
+             </ol>
+             </li>
+            <li> Return <var>p</var></li>
+            </ol>
+            <div>
+              <p class="note"> <var>resultList</var> MAY be ordered by the UA so
+              that MIME types closer to the beginning have a larger probability
+              of being supported by encode accelerators.</p>
```

Comment:
Is hardware support the only reason one might promote something to the beginning, and what if there are multiple codecs with hardware support? Can you bake this advice into the normative part of  the algorithm (keyword MAY is normative) and say that `resultList` must be ordered such that the formats that the UA would prefer the web application to use are at the front, for whatever reason?